index.d.ts

```diff
@@ -0,0 +1,136 @@
+type KeyEffectDistance = {
+  distanceX: number;
+  distanceY: number;
+};
+
+interface KeyEffect {
+  initialX: number;
+  initialY: number;
+  name?: string;
+  description?: string;
+}
+
+interface KeyMap<E> {
+  keys: Record<string, E>;
+  name?: string;
+  description?: string;
+}
+
+interface KeyMappingConfig {
+  dpad?: KeyMap<KeyEffect & KeyEffectDistance>[];
+  tap?: KeyMap<KeyEffect>[];
+  swipe?: KeyMap<KeyEffect & KeyEffectDistance>[];
+}
+
+type DeviceRendererKeyMapping = {
+  enable(isEnable: boolean): void;
+  setConfig(config: KeyMappingConfig): void;
+  activeKeyMappingDebug(isTraceActivate?: boolean, isGridActivate?: boolean): void;
+};
+
+type VmEvent = 'beforeunload' | 'fingerprint' | 'gps' | 'BATTERY_LEVEL' | string // TODO Provide an exhaustive list
+
+type VmCommunication = {
+  disconnect(): void;
+  addEventListener(event: VmEvent, callback: (msg: string) => void): void; // TODO Verify if msg is always string
```

Comment:
TODO

Reply:
From what I've found in the code for now, all values for `msg` where strings